File: JavaScript/1695-Maximum Erasure Value.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maximumUniqueSubarray = function(nums) {
    let left = 0, currSum = 0, res = 0;
    let used = new Set();
    
    for (let i=0; i<nums.length; i++){
        if (used.has(nums[i])){
            res = Math.max(res, currSum);
            while (used.has(nums[i])){
                used.delete(nums[left]);
                currSum -= nums[left];
                left += 1;                
            }
        }
        currSum += nums[i];
        used.add(nums[i]);
    }
    
    res = Math.max(res, currSum);
    return res;
};

File: C++/1695-Maximum Erasure Value.cpp
class Solution {
public:
    int maximumUniqueSubarray(vector<int>& nums) {
        int res = 0, left = 0, N = nums.size(), curr = 0;
        vector<int> cnt(10001, 0);

        for (int i=0; i<N; i++) {
            curr += nums[i];
            cnt[nums[i]]++;
            while (cnt[nums[i]] > 1) {
                cnt[nums[left]]--;
                curr -= nums[left];
                left++;
            }
            res = max(res, curr);
        }

        return res;
    }
};